### PR TITLE
chore: Bump to 24.7-SNAPSHOT

### PR DESCRIPTION
**Pull request by Artur-**

### CODE REVIEW

File: packages/java/gradle-plugin/src/functionalTest/kotlin/com/vaadin/hilla/gradle/plugin/AbstractGradleTest.kt [hunk 1]
```diff
@@ -25,7 +25,7 @@ import org.junit.Before
  */
 abstract class AbstractGradleTest {
 
-    val hillaVersion = System.getenv("hilla.version").takeUnless { it.isNullOrEmpty() } ?: "24.6-SNAPSHOT"
+    val hillaVersion = System.getenv("hilla.version").takeUnless { it.isNullOrEmpty() } ?: "24.7-SNAPSHOT"
```
Comment (taefi) [inline packages/java/gradle-plugin/src/functionalTest/kotlin/com/vaadin/hilla/gradle/plugin/AbstractGradleTest.kt:28]: Is there a reason that flow shouldn't be updated to `24.7-SNAPSHOT`?

File: packages/java/tests/gradle/single-module/gradle.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
 org.gradle.daemon=false
-hillaVersion=24.6-SNAPSHOT
+hillaVersion=24.7-SNAPSHOT
```
Comment (taefi) [inline packages/java/tests/gradle/single-module/gradle.properties:2]: Is there a reason that flow shouldn't be updated to `24.7-SNAPSHOT`?

**Review by taefi — COMMENTED** [review]
Apart from the Flow version, I'd suggest to also update the npm versions to 24.7.0-alpha0 (similar to what we did for 24.6) so that it doesn't just work by pulling in the old versions unintentionally.

**Comment by ZheSun88** [comment]
can you make the necessary changes @taefi

File: packages/java/gradle-plugin/src/functionalTest/kotlin/com/vaadin/hilla/gradle/plugin/AbstractGradleTest.kt (hunk 1 above)
Comment (Artur-) [reply packages/java/gradle-plugin/src/functionalTest/kotlin/com/vaadin/hilla/gradle/plugin/AbstractGradleTest.kt:28]: No, but it is not related to this PR. It has some issues as you can see in #2961

File: packages/java/tests/gradle/single-module/gradle.properties (hunk 2 above)
Comment (Artur-) [reply packages/java/tests/gradle/single-module/gradle.properties:2]: No, but it is not related to this PR. It has some issues as you can see in #2961

**Comment by Artur-** [comment]
Feel free to update the npm versions, in this PR or a separate one

**Comment by taefi** [comment]
> can you make the necessary changes @taefi

I can, but it has to wait until I'm done with another build failure.

**Comment by Artur-** [comment]
This does not break anything so why don't we merge this first, before some other PR cause conflicts with this?

**Comment by Artur-** [comment]
If it is only about running `npm version 24.7.0-alpha0` in each folder, I can do that..

**Comment by Artur-** [comment]
Updated, let's see what happens..

**Review by taefi — APPROVED** [review]

**Pull request merged**